### DISCUSSION TITLE
Question about the benchmarks

### DISCUSSION THREAD

**Discussion by sanikolaev** [siglens/siglens#474] [Q&A] [answered]
Hi

I have a question regarding the benchmarks posted on [Siglens' Blog about being 54x faster than Clickhouse](https://www.siglens.com/blog/siglens-54x-faster-than-clickhouse.html) and [Siglens' Blog about being 1025x faster than Elasticsearch](https://www.siglens.com/blog/siglens-1025x-faster-than-elasticsearch.html).

Am I missing something, or does this query:
```
curl -X POST -d '{
    "searchText": "SELECT airport_fee, count(*) FROM trips GROUP BY airport_fee",
    "indexName": "trips",
    "startEpoch": "now-24h",
    "endEpoch": "now",
    "queryLanguage": "SQL"
}' http://localhost:5122/api/search | python3 -m json.tool
```

filter out all the data except for the last 24 hours, while this

```
SELECT airport_fee, count(*) FROM trips GROUP BY airport_fee
```

does not?

If this is correct, are there benchmark results for Siglens without this filtering?

**Comment by AndrewHess** [comment, marked as answer]
You're right, the first query takes 24 hours of data and the second query takes all the data. However, when we ran these benchmarks we ran the queries within 24 hours of starting the ingestion process and we also verified that the results returned by each corresponding set of queries was the same.

**Comment by yusufozturk** [comment]
@AndrewHess, so your benchmarks are biased...

For the ClickHouse tests, you did not create partitioning for the ingested data, which means all the data was likely ingested into a single partition. Even though ClickHouse may use default ordering based on epoch, not using partitioning can lead to slower query response times.

In your tests, you wrote the data into daily partitions, so time range queries scanned a smaller amount of data. This creates an unfair comparison, as ClickHouse wasn’t given the same opportunity to optimize for time-based queries.

Additionally, you should always apply epoch filtering in ClickHouse to make searches much faster. Without this, ClickHouse has to scan the entire dataset to return results, which naturally slows down query performance.

On the other hand, you used a time filter in your queries, which allowed you to fetch data from a smaller subset, further tilting the comparison in your favor. This is not a fair test.

If you believe your solution is truly faster than ClickHouse, I encourage you to submit a benchmark on ClickBench, which is open source and welcomes all benchmark reports. If your solution can’t handle many of the queries tested there, how do you claim it can replace Splunk? Selecting just four queries for a benchmark where you have an advantage due to a specially designed setup only casts doubt on your claims.

I look forward to seeing your ClickBench PR.

/cc @alexey-milovidov
